src/binding/python/openpmd_api/DaskDataFrame.py

```diff
@@ -0,0 +1,83 @@
+"""
+This file is part of the openPMD-api.
+
+Copyright 2021 openPMD contributors
+Authors: Axel Huebl, Dmitry Ganyushin
+License: LGPLv3+
+"""
+import numpy as np
+try:
+    import dask.dataframe as dd
+    from dask.delayed import delayed
+    found_dask = True
+except ImportError:
+    found_dask = False
+try:
+    import pandas as pd
+    found_pandas = True
+except ImportError:
+    found_pandas = False
+
+
+def particles_to_daskdataframe(particle_species):
+    """
+    Load all records of a particle species into a Dask DataFrame.
+
+    Parameters
+    ----------
+    particle_species : openpmd_api.ParticleSpecies
+        A ParticleSpecies class in openPMD-api.
+
+    Returns
+    -------
+    dask.dataframe
+        A dask dataframe with particles as index and openPMD record
+        components of the particle_species as columns.
+
+    Raises
+    ------
+    ImportError
+        Raises an exception if dask or pandas are not installed
+
+    See Also
+    --------
+    openpmd_api.BaseRecordComponent.available_chunks : available chunks that
+        are used internally to parallelize particle processing
+    dask.dataframe : the central dataframe object created here
+    """
+    if not found_dask:
+        raise ImportError("dask NOT found. Install dask for Dask DataFrame "
+                          "support.")
+    if not found_pandas:
+        raise ImportError("pandas NOT found. Install pandas for DataFrame "
+                          "support.")
+
+    # get optimal chunks: query first non-constant record component and
+    #                     assume the same chunking applies for all of them
+    #                     in a particle species
+    chunks = None
+    for k_r, r in particle_species.items():
+        for k_rc, rc in r.items():
+            if not rc.constant:
+                chunks = rc.available_chunks()
```

Comment:
This assumes that chunks are equal across components. What happens if they're not? Will things just be less efficient or will things not work? In the latter case, we should probably guard for this case and throw an error.

Reply:
Yep, will just be less efficient.